daemon/src/model/cfd.rs

```diff
@@ -598,13 +599,48 @@ impl Cfd {
         ))
     }
 
-    pub fn start_rollover(
-        &self,
-    ) -> Result<(Event, (RolloverParams, Dlc, Duration)), CannotRollover> {
+    pub fn start_rollover_taker(&self) -> Result<(Event, (RolloverParams, Dlc))> {
+        anyhow::ensure!(!self.during_rollover && self.role == Role::Taker);
+
         self.can_rollover()?;
 
         Ok((
             Event::new(self.id, CfdEvent::RolloverStarted),
+            (
+                RolloverParams::new(
+                    self.initial_price,
+                    self.quantity,
+                    self.leverage,
+                    self.refund_timelock_in_blocks(),
+                    1, // TODO: Where should I get the fee rate from?
+                ),
+                self.dlc.as_ref().ok_or(CannotRollover::NoDlc)?.clone(),
+            ),
+        ))
+    }
+
+    pub fn receive_rollover_proposal(self, proposal: RolloverProposal) -> Result<Event> {
```

Comment:
I am liking this!

Reply:
Good to get some more confirmation, because I wasn't sure if these state checking methods that just emit events were what we wanted necessarily.